File: CSharpDefinitions/Samples/AxisSettings.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace CSharpDefinitions.Samples.AxisSettings {

    public interface dxChartOptions {

        ChartCommonAxisSettings commonAxisSettings { get; }
    }

    public interface dxPolarChartOptions {

        PolarCommonAxisSettings commonAxisSettings { get; }
    }

    public interface ChartAxis : ChartCommonAxisSettings, Axis { }

    public interface PolarAxis : PolarCommonAxisSettings, Axis { }

    public interface ChartCommonAxisSettings {

        ChartAxisLabel label { get; }
    }

    public interface PolarCommonAxisSettings {

        PolarAxisLabel label { get; }
    }

    [InjectedType]
    public interface ChartAxisLabel : CommonAxisLabelSettings {

        string alignment { get; }
    }

    [InjectedType]
    public interface PolarAxisLabel : CommonAxisLabelSettings {

        [DefaultValue("enlargeTickInterval")]
        string overlappingBehavior { get; }
    }

    [InjectedType]
    public interface Axis {

        [DefaultValue(10)]
        int logarithmBase { get; }
    }

    [InjectedType]
    public interface CommonAxisLabelSettings {

        [DefaultValue(5)]
        int indentFromAxis { get; }
    }
}

File: JSDoc/Samples/AxisSettings.js
/**
 * @class
 */
var dxChartOptions = {

    /**
     * @memberof dxChartOptions
     * @type {ChartCommonAxisSettings}
     */
    commonAxisSettings
};

/**
 * @class
 */
var dxPolarChartOptions = {

    /**
     * @memberof dxPolarChartOptions
     * @type {PolarCommonAxisSettings}
     */
    commonAxisSettings
};

/**
 * @class ChartAxis
 * @extends ChartCommonAxisSettings
 * @mixes Axis
 */

/**
 * @class PolarAxis
 * @extends PolarCommonAxisSettings
 * @mixes Axis
 */

/**
 * @mixin
 */
var Axis = {

    /**
     * @memberof Axis
     * @type {int}
     * @default 10
     */
    logarithmBase
};

/**
 * @class
 */
var ChartCommonAxisSettings = {

    /**
     * @memberof ChartCommonAxisSettings
     * @mixes Label
     * @type {object}
     */
    label: {
        /**
         * @memberof ChartCommonAxisSettings.label
         * @type {string}
         */
        alignment
    }
};

/**
 * @class
 */
var PolarCommonAxisSettings = {

    /**
     * @memberof PolarCommonAxisSettings
     * @mixes Label
     * @type {object}
     */
    label: {
        /**
         * @memberof PolarCommonAxisSettings.label
         * @type {string}
         * @default enlargeTickInterval
         */
        overlappingBehavior
    }
};

/**
 * @mixin
 */
var Label = {

    /**
     * @memberof Label
     * @type {int}
     * @default 5
     */
    indentFromAxis,
};